python/lsst/butler/registry.py

```diff
@@ -30,18 +30,28 @@
 from .quantum import Quantum
 from .units import DataUnit, DataUnitTypeSet
 from .storageClass import StorageClass
+from .config import Config
 
+class RegistryConfig(Config):
+    pass
 
 class Registry:
     """Basic SQL backed registry.
     """
 
-    def __init__(self, dbname='sqlite:///:memory:', id=0):
-        self.engine = create_engine(dbname)
-        self.id = id
+    def __init__(self, config):
+        self.config = RegistryConfig(config)['registry']
+        self.engine = create_engine(self.config['dbname'])
+        self.id = self.config['id']
 
         metadata.create_all(self.engine)
 
+    @staticmethod
+    def fromConfig(config):
+        from lsst.daf.persistence import doImport
```

Comment:
I'm not really a fan of these deferred imports as it hides the dependencies.

Reply:
So I moved the import of `doImport` to the top. Of course it is still doing a deferred import, since that is the whole point of `doImport`. But I assume that is not what you were objecting to.

Reply:
Yes, it was the `from` I wanted moved up top.